torchdynamo_poc/torchbench.py

```diff
@@ -67,18 +102,31 @@ def forward(*inputs):
         result = loaded_module.forward(*inputs)
         result = tuple() if result is None else result
         return (result,) if was_unwrapped else result
-
     return forward
 
 
-def run(func, num_iter, compiler):
-    """Run a function a number of times and print out how long eveything took."""
-    start_time = time.time_ns()
-    for _i in range(num_iter):
-        with torchdynamo.optimize(compiler):
-            func()
-    end_time = time.time_ns()
-    print(f"Finished in {end_time - start_time} ns")
+def run(func: Callable[[], List[torch.Tensor]], num_iter):
+    """Run a function a number of times."""
+    results = []
+    for _ in range(num_iter):
+        results += func()
+    return results
+
+
+def check_results(compiled_results, eager_results):
+    for compiled_result, eager_result in zip(compiled_results, eager_results):
+        if abs(torch.mean(compiled_result) - torch.mean(eager_result)) > 0.001:
+            print("Compiled result does not match eager result")
+            return
+    print("Compiled result matches eager result!")
+
+
+def print_time_stats(times, *, warmup_iters: int = 0):
+    iter_times = torch.tensor(times[warmup_iters:])
+    print(f"Mean: {torch.mean(iter_times.to(float))} ns")
```

Comment:
nit: mean and std are not very useful for performance benchmarking (in performance benchmarking we are usually interested in the distribution of times -- ideally with plots of the distribution). I would recommend you print the min, 5, 50, 95, max percentiles.

Reply:
That makes sense, thanks! I'll add it in a follow up PR